fix(tests): migrate whitespace mutations to non-whitespace syntax and ban the whitespace syntax

adnrs96 opened the pull request:
This is a prepratory PR for the changes to come in the effor to achieve #1158 
This basically migrates all of the existing tests which use whitespace mutation syntax to not use it.

Moving forward, we will ban this syntax in an upcoming PR and add tests to ensure that it stays banned but this PR is just about preparring for that effort.


EDIT: Due to codecoverage failing for just migrating the tests, I decided to make this PR about the entire #1158 issue. Now this PR has commits which ban whitespace syntax for mutations by always expecting a service.

Fixes: #1158.

adnrs96 commented:
Loss in code coverage will get fixed the the PR which blocks whitespace syntax.
I created this PR so that the lots of test remains separate but now I feel maybe I should haven't seperated out since these test modifications aren't totally isolated from the other code as the code coverage loss suggests.

adnrs96 commented:
Seems like github is suffering from somekinda issue since pushing to my repo isn't adding commits to the PR!

adnrs96 commented on storyscript/compiler/semantics/ExpressionResolver.py:
```diff
@@ -580,9 +579,12 @@ def service(self, tree):
+        # variable exists -> mutation/event-based service
+        if t.type() == ObjectType.instance():
+            # In case of event-based service return Anytype for now.
+            return base_symbol(AnyType.instance())
```
We will address this when fixing the issue #1166 
By this I mean always returning the AnyType

wilzbach commented on storyscript/compiler/semantics/ExpressionResolver.py:
```diff
@@ -580,15 +579,16 @@ def service(self, tree):
+            # In case of event-based service return Anytype for now.
+            return base_symbol(AnyType.instance())
+
+        tree.expect(0, 'service_name_expected', found='variable')
```
I would recommend to include the variable name in the message instead of just `variable`.

wilzbach commented on storyscript/compiler/semantics/TypeResolver.py:
```diff
@@ -225,13 +225,10 @@ def when_block(self, tree, scope):
+
+        # Whitespace syntax for mutations is not allowed anymore.
+        tree.expect(name is None or isinstance(name.type(), ObjectType),
+                    'service_name_expected', found='variable')
```
Maybe print the actual variable and improve the error message to make it clear to the user that he accidentally used an existing variable.

adnrs96 commented:
Thanks for reviewing @wilzbach 
I addressed your comments and added a commit to improve upon the error msg experience.
PR has been updated to represent that :)

adnrs96 commented:
Also I would like to mention here that for cases like 
```
twitter stream as client
    when client tweet track:"Star Trek" as result
        result retweet with_content: "yeah!"
```

Error would look like 
```
Error: syntax error in test.story at line 3, column 9

3|            result retweet with_content: "yeah!"
              ^^^^^^

E0148: Service name expected but existing variable `result` found.
```

Basically for case of twitter the event returns a map instead of a `ServiceOutput` and wrongfully using it as a service should and will error out.
Its just that the error would say `result` to be a variable which is kinda true. Right now inside the nested block you can even reassign this `result` and I believe it should be alright to call it a `variable`?

wilzbach commented on storyscript/ErrorCodes.py:
```diff
@@ -266,6 +264,14 @@ class ErrorCodes:
+        'E0148',
+        'Service name expected but existing variable `{var}` found.'
+    )
+    inline_ser_expected_ser_name = (
```
No hard to understand abbreviations like `ser` please. Use `service` it's not that much longer.
Also start with `service` for better grouping like e.g. `service_name_not_var`.

adnrs96 replied:
will do that :+1:
